artifacts/registry.py

```diff
@@ -33,6 +77,25 @@ def DeregisterDefinition(self, artifact_definition):
 
     del self._artifact_definitions[artifact_definition_name]
 
+  @classmethod
+  def DeregisterSourceType(cls, source_type_class):
```

Comment:
having a deregister without a register is confusing. Perhaps the better approach is to have

DEFAULT_TYPE_CLASSES

and in __init__ call a register method that registers them into the class dict.